src/scaffold.ts

```diff
@@ -31,31 +32,53 @@ type MessageResponse<MessageType extends string> = Awaited<
   ReturnType<Extract<MessageSender, (type: MessageType, body: any) => any>>
 >;
 
-const scaffoldWebviewCache = new WebviewPanelCache();
+interface PrefilledTemplateOptions {
+  templateName?: string;
+  [key: string]: string | undefined;
+}
 
+const scaffoldWebviewCache = new WebviewPanelCache();
+const logger = new Logger("scaffold");
 export function registerProjectGenerationCommands(): vscode.Disposable[] {
   return [
     registerCommandWithLogging("confluent.scaffold", scaffoldProjectRequest),
-    registerCommandWithLogging("confluent.resources.scaffold", scaffoldProjectRequest),
+    registerCommandWithLogging("confluent.resources.scaffold", resourceScaffoldProjectRequest),
   ];
 }
 
-export const scaffoldProjectRequest = async (item?: KafkaCluster | KafkaTopic) => {
+async function resourceScaffoldProjectRequest(item?: KafkaCluster | KafkaTopic) {
+  if (item instanceof KafkaCluster) {
+    return await scaffoldProjectRequest({ cc_bootstrap_server: item.bootstrapServers });
+  } else if (item instanceof KafkaTopic) {
+    const cluster = await getResourceManager().getClusterForTopic(item);
+    return await scaffoldProjectRequest({
+      cc_bootstrap_server: cluster?.bootstrapServers,
+      cc_topic: item.name,
+    });
+  }
```

Comment:
Before I forget, we may want to duplicate value(s) since they may match multiple template option keys/fields:
```suggestion
  if (item instanceof KafkaCluster) {
    return await scaffoldProjectRequest({
      bootstrap_server: item.bootstrapServers,
      cc_bootstrap_server: item.bootstrapServers,
    });
  } else if (item instanceof KafkaTopic) {
    const cluster = await getResourceManager().getClusterForTopic(item);
    return await scaffoldProjectRequest({
      bootstrap_server: cluster?.bootstrapServers,
      cc_bootstrap_server: cluster?.bootstrapServers,
      cc_topic: item.name,
      topic: item.name
    });
  }
```